core/src/main/java/com/google/cloud/sql/ConnectorRegistry.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql;
+
+import com.google.cloud.sql.core.CoreSocketFactory;
+
+/** The public java API to configure the connector registry. */
+public enum ConnectorRegistry {
+  INSTANCE;
+
+  /** Returns the singleton instance of the Connector. */
+  public static Connector getConnector() {
```

Comment:
This is not a "connector" it's a `getInstance` or `getRegistry`.

Reply:
This matches Alloy DB JDBC Connector's [ConnectorRegistry.getConnector()](https://github.com/GoogleCloudPlatform/alloydb-java-connector/blob/8496eecee4787a6e912973bcbe4bfa2255732c5b/alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/ConnectorRegistry.java#L62C4-L64C4)

Reply:
The interface, yes. But the CoreSocketFactory is more like AlloyDB java's ConnectorRegistry. I think we should just treat the CoreSocketFactory as an equivalent to the registry and not introduce a wrapper class here.